src/server/schema.js

```diff
@@ -58,55 +64,115 @@ const getQuerySchemaForType = (esType) => {
     ): [${esTypeObjName}]`;
 };
 
+const getFieldGQLTypeMapForProperties = (esInstance, esIndex, properties) => {
+  const result = Object.keys(properties).map((field) => {
+    const esFieldType = properties[field].type;
+    const gqlType = getGQLType(esInstance, esIndex, field, esFieldType);
+    return {
+      field, type: gqlType, esType: esFieldType, props: properties[field].properties,
+    };
+  });
+  return result;
+};
+
 const getFieldGQLTypeMapForOneIndex = (esInstance, esIndex) => {
   const fieldESTypeMap = esInstance.getESFieldTypeMappingByIndex(esIndex);
-  const fieldGQLTypeMap = Object.keys(fieldESTypeMap).map((field) => {
-    const esFieldType = fieldESTypeMap[field];
-    const gqlType = getGQLType(esInstance, esIndex, field, esFieldType);
-    return { field, type: gqlType };
+  return getFieldGQLTypeMapForProperties(esInstance, esIndex, fieldESTypeMap);
+};
+
+const getArgsByField = (fieldType, props) => {
+  const keys = Object.keys(props);
+  const baseProps = [];
+  keys.forEach((k) => {
+    if (props[k].type !== 'nested') baseProps.push({ field: k, type: esgqlTypeMapping[props[k].type] });
   });
-  return fieldGQLTypeMap;
+  return `(${baseProps.map((entry) => `${entry.field}: [${entry.type}]`).join(',')})`;
```

Comment:
I think for nested fields we don't need them in args, should be fine if we just remove this?